src/control/cmd/dmg/pool.go

```diff
@@ -149,158 +150,191 @@ func (c *PoolCreateCmd) Execute(args []string) error {
 			resp.UUID, formatPoolSvcReps(resp.SvcReps), resp.NumRanks)
 	}
 
-	c.log.Infof("Pool-create command %s\n", msg)
+	cmd.log.Infof("Pool-create command %s\n", msg)
 
 	return err
 }
 
-// PoolDestroyCmd is the struct representing the command to destroy a DAOS pool.
-type PoolDestroyCmd struct {
+// poolCmd is the base struct for all pool commands that work with existing pools.
+type poolCmd struct {
 	logCmd
-	ctlInvokerCmd
 	jsonOutputCmd
+	ctlInvokerCmd
+	ID   string `long:"pool" required:"1" description:"Unique ID of DAOS pool"`
+	UUID string
+}
+
+// resolveID attempts to resolve the supplied pool ID into a UUID.
+func (cmd *poolCmd) resolveID() error {
+	if _, err := uuid.Parse(cmd.ID); err == nil {
+		cmd.UUID = cmd.ID
+		return nil
+	}
+
+	if cmd.ID == "" {
+		return errors.New("no pool ID supplied")
+	}
+
```

Comment:
Any reason not to check this first, before trying to parse a UUID?

Reply:
Hmm, this is a good point. I added the check for an empty ID later while adding unit tests. No reason it couldn't go first, and would be a slight optimization.